ministryofjustice/correspondence_tool_staff
# CT-2979 edit closure details of offender SARs

## Description

Adds the ability for users to edit the date of a case closure for Offender SARs after the case is closed. Solution utilises the existing `*_step` pattern for implementation.

## Self-review checklist


### Screenshots


#### Before
<img width="987" alt="Screenshot 2020-08-12 at 14 08 16" src="https://user-images.githubusercontent.com/13377553/90018777-5a3ec880-dca5-11ea-8188-44646e111a74.png">

#### After
<img width="993" alt="Screenshot 2020-08-12 at 14 09 19" src="https://user-images.githubusercontent.com/13377553/90018834-717db600-dca5-11ea-8427-704e4868199c.png">

##### Edit page:
<img width="966" alt="Screenshot 2020-08-12 at 14 10 06" src="https://user-images.githubusercontent.com/13377553/90018902-8e19ee00-dca5-11ea-9ae0-85e1702592f9.png">

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2979

### Deployment
N/A

### Manual testing instructions
1. Log in as a Branston user
2. Progress a case to a closed state
3. On show page -> click the 'Change' link next to 'Final deadline'
4. Add a updated date for the 'Date responded'
5. Click continue
6. New date should be reflected in the 'Date response sent to requester' field.
